src/core/io/curlutils.cpp

```diff
@@ -0,0 +1,512 @@
+                   "request.";
+        case Loop_Detected:
+            return "The server detected an infinite loop while processing the request (sent "
+                   "instead of 208 Already Reported).";
+        case Bandwidth_Limit_Exceeded:
+            return "The server has exceeded the bandwidth specified by the server administrator; "
+                   "this is often used by shared hosting providers to limit the bandwidth of "
+                   "customers.";
+        case Not_Extended:
+            return "Further extensions to the request are required for the server to fulfil it.";
+        case Network_Authentication_Required:
+            return "The client needs to authenticate to gain network access.";
+        case Unknown_Error:
+            return "The 520 error is used as a \"catch-all response for when the origin server "
+                   "returns something unexpected\", listing connection resets, large headers, and "
+                   "empty or invalid responses as common triggers.";
+        case Web_Server_Is_Down:
+            return "The origin server has refused the connection from Cloudflare.";
+        case Connection_Timed_Out:
+            return "Cloudflare could not negotiate a TCP handshake with the origin server.";
+        case Origin_Is_Unreachable:
+            return "Cloudflare could not reach the origin server; for example, if the DNS records "
+                   "for the origin server are incorrect.";
+        case A_Timeout_Occurred:
+            return "Cloudflare was able to complete a TCP connection to the origin server, but did "
+                   "not receive a timely HTTP response.";
+        case SSL_Handshake_Failed:
+            return "Cloudflare could not negotiate a SSL/TLS handshake with the origin server.";
+        case Invalid_SSL_Certificate:
+            return "Used by Cloudflare and Cloud Foundry's gorouter to indicate failure to "
+                   "validate the SSL/TLS certificate that the origin server presented.";
+        case Railgun_Listener_to_Origin_Error:
+            return "Error 527 indicates that the request timed out or failed after the WAN "
+                   "connection had been established.";
+        case Origin_DNS_Error:
+            return "Error 530 indicates that the requested host name could not be resolved on the "
+                   "Cloudflare network to an origin server.";
+        case Network_Read_Timeout_Error:
+            return "Used by some HTTP proxies to signal a network read timeout behind the proxy to "
+                   "a client in front of the proxy.";
+    }
+    return "Unknown response code";
+}
+
+std::filesystem::path downloadAndCacheIfUrl(const std::filesystem::path& url) {
+    if (!net::isUrl(url)) {
+        return url;
+    }
+
+    auto cacheDir = filesystem::getPath(PathType::Cache);
+    std::filesystem::create_directories(cacheDir);
```

Comment:
Would make sense to use the downloads directory instead. Aligns better with how most apps handle downloads.

Reply:
no, see above